app/src/sharedTest/java/org/oppia/android/app/databinding/DrawableBindingAdaptersTest.kt

```diff
@@ -0,0 +1,178 @@
+import org.oppia.android.domain.classify.rules.fractioninput.FractionInputModule
+import org.oppia.android.domain.classify.rules.imageClickInput.ImageClickInputModule
+import org.oppia.android.domain.classify.rules.itemselectioninput.ItemSelectionInputModule
+import org.oppia.android.domain.classify.rules.multiplechoiceinput.MultipleChoiceInputModule
+import org.oppia.android.domain.classify.rules.numberwithunits.NumberWithUnitsRuleModule
+import org.oppia.android.domain.classify.rules.numericinput.NumericInputRuleModule
+import org.oppia.android.domain.classify.rules.ratioinput.RatioInputModule
+import org.oppia.android.domain.classify.rules.textinput.TextInputRuleModule
+import org.oppia.android.domain.onboarding.ExpirationMetaDataRetrieverModule
+import org.oppia.android.domain.oppialogger.LogStorageModule
+import org.oppia.android.domain.oppialogger.loguploader.LogUploadWorkerModule
+import org.oppia.android.domain.oppialogger.loguploader.WorkManagerConfigurationModule
+import org.oppia.android.domain.question.QuestionModule
+import org.oppia.android.domain.topic.PrimeTopicAssetsControllerModule
+import org.oppia.android.testing.TestImageLoaderModule
+import org.oppia.android.testing.TestLogReportingModule
+import org.oppia.android.testing.robolectric.RobolectricModule
+import org.oppia.android.testing.threading.TestDispatcherModule
+import org.oppia.android.testing.time.FakeOppiaClockModule
+import org.oppia.android.util.accessibility.AccessibilityTestModule
+import org.oppia.android.util.caching.testing.CachingTestModule
+import org.oppia.android.util.gcsresource.GcsResourceModule
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.parser.html.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.image.ImageParsingModule
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Tests for [DrawableBindingAdapters]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(
+  application = DrawableBindingAdaptersTest.TestApplication::class,
+  qualifiers = "port-xxhdpi"
+)
+class DrawableBindingAdaptersTest {
+
+  @Inject
+  lateinit var context: Context
+
+  private var resourceColor: Int = 0
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+    resourceColor = context.resources.getColor(R.color.chapterCardShadow, null)
+  }
+
+  private fun setUpTestApplicationComponent() {
+    ApplicationProvider.getApplicationContext<TestApplication>()
+      .inject(this)
+  }
+
+  @Test
+  fun testDrawableBindingAdapters_setBackgroundColor_hasCorrectBackgroundColor() {
+    val v = View(context)
+    setBackgroundColor(v, resourceColor)
+
+    val color = (v.getBackground() as ColorDrawable).color
+    assertThat(color).isEqualTo(
+      0xff000000.toInt() or resourceColor
```

Comment:
Shouldn't we make the assertion to match with the unique color?
E.g- It looks like that the test case will pass in such cases where both correspond to different values.